### PR TITLE
Z index registry / clean-up

### PR DESCRIPTION
**Pull request by rusackas**
### CATEGORY

Choose one

- [ ] Bug Fix
- [ ] Enhancement (new features, refinement)
- [x] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
This PR aims to simplify and consolidate the use of z-index values (z-indices?) throughout the codebase. 

In the main `stylesheets/less/variables.less`, there are a few semantic layers onto which you can place elements. Add more if you want, but try to keep it minimal. Think of them like layers in Illustrator similar. 

I found a handful of z-index values that appeared to make no difference, so I've removed them. This is the riskiest part of this PR, but for each of these, I poked around trying to find the element(s) in the UI and make sure that it looked the same before/after removing the value. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
n/a - everything should look identical, unless there's an instance of an element I missed that becomes obscured by being covered up.

### REVIEWERS
@williaster (some of the z-index values I removed were yours, according to git — I hope I didn't break anything you added for good reason)
@mistercrunch (you probably have some context about these elements too, and can make sure I didn't do anything stupid)

### CODE REVIEW

File: superset/assets/stylesheets/less/variables.less
Comment (etr2460) [inline superset/assets/stylesheets/less/variables.less]: why not `dashboard` instead of `dash`?

File: superset/assets/stylesheets/less/variables.less
Comment (etr2460) [inline superset/assets/stylesheets/less/variables.less]: Let's remove the `Try to` here

File: superset/assets/src/messageToasts/stylesheets/toast.less
Comment (etr2460) [inline superset/assets/src/messageToasts/stylesheets/toast.less]: comment probably isn't required now right?

File: superset/assets/stylesheets/less/variables.less
Comment (etr2460) [inline superset/assets/stylesheets/less/variables.less]: any reason you got rid of `index` in these var names? I kinda liked it in there

**Review by etr2460 — COMMENTED** [review]
some variable name comments, i'll try to pull and test as well, probably early next week

File: superset/assets/stylesheets/less/variables.less
Comment (rusackas) [reply superset/assets/stylesheets/less/variables.less]: I was just trying to be terse... I'll make it dashboards, that's fine.

File: superset/assets/stylesheets/less/variables.less
Comment (rusackas) [reply superset/assets/stylesheets/less/variables.less]: There is no try

File: superset/assets/src/messageToasts/stylesheets/toast.less
Comment (rusackas) [reply superset/assets/src/messageToasts/stylesheets/toast.less]: quite right! Thought I nabbed that.

File: superset/assets/stylesheets/less/variables.less
Comment (rusackas) [reply superset/assets/stylesheets/less/variables.less]: Just felt so dash-tastic with them in there! I figured just `z` captured the intent enough.

That said, I don't have a strong opinion. `z-index-...` was my first notion, so I'm glad to hear you liked it... I'll put it back!

**Comment by mistercrunch** [comment]
@etr2460 we'd ❤️ your blessing before we merge this

**Comment by etr2460** [comment]
Sorry for the delay, was wrestling with my dev environment this week. I did some smoke tests and everything looks good to me!

**Review by etr2460 — APPROVED** [review]

**Review by mistercrunch — APPROVED** [review]

**Comment by mistercrunch** [comment]
Solved the 2 minor conflicts in Github UI, will merge once CI passes

**Pull request merged**
